tests/tsutil.py

```diff
@@ -0,0 +1,153 @@
+    is set to False, then empty metadata fields in the returned tree sequence will be
+    set to ``b'null'`` to ensure all metadata is valid JSON.
+
+    For testing purposes, we can set ``skip_last`` to check what happens if we have
+    some samples that are not associated with an individual in the tree sequence.
+    """
+    ts = msprime.simulate(
+        n * ploidy,
+        recombination_rate=1,
+        mutation_rate=mutation_rate,
+        length=length,
+        random_seed=100,
+    )
+    tables = ts.dump_tables()
+    tables.populations.clear()
+    no_metadata = b"{}" if strict_json_metadata else b""  # Fix when schemas in tsinfer
+    for i in range(n):
+        individual_meta = no_metadata
+        pop_meta = no_metadata
+        location = [i, i]
+        if i % 2 == 0:
+            # Add unicode metadata to every other individual: 8544+i = Roman numerals
+            individual_meta = json.dumps({"unicode id": chr(8544 + i)}).encode()
+            # TODO: flags should use np.iinfo(np.uint32).max. Change after solving issue
+            # https://github.com/tskit-dev/tskit/issues/1027
+            individual_flags = np.random.randint(0, np.iinfo(np.int32).max)
+            # Also for populations: chr(127462) + chr(127462+i) give emoji flags
+            pop_meta = json.dumps({"utf": chr(127462) + chr(127462 + i)}).encode()
+        tables.populations.add_row(metadata=pop_meta)  # One pop for each individual
+        if i < n - 1 or skip_last is False:
+            tables.individuals.add_row(individual_flags, location, individual_meta)
+
+    node_metadata = []
+    node_populations = tables.nodes.population
+    for node in ts.nodes():
+        if node.is_sample():
+            print(node.id, ploidy, node.id // ploidy)
+            node_populations[node.id] = node.id // ploidy
+        if node.id % 3 == 0:  # Scatter metadata into nodes: once every 3rd row
+            node_metadata.append(json.dumps({"node id": node.id}).encode())
+        else:
+            node_metadata.append(no_metadata)
+    tables.nodes.population = node_populations
+    tables.nodes.packset_metadata(node_metadata)
+
+    site_metadata = []
+    for site in ts.sites():
+        if site.id % 4 == 0:  # Scatter metadata into sites: once every 4th row
+            site_metadata.append(json.dumps({"id": f"site {site.id}"}).encode())
+        else:
+            site_metadata.append(no_metadata)
+    tables.sites.packset_metadata(site_metadata)
+
+    nodes_individual = tables.nodes.individual  # Assign individuals to sample nodes
+    sample_individuals = np.repeat(
+        np.arange(n, dtype=tables.nodes.individual.dtype), ploidy
+    )
+    if skip_last:
+        # Should work even if some sample nodes are not assigned to an individual
+        sample_individuals[sample_individuals == n - 1] = tskit.NULL
+    nodes_individual[ts.samples()] = sample_individuals
+    tables.nodes.individual = nodes_individual
+    print(tables.populations)
```

Comment:
print

Reply:
Damn - I need something to take these stray debugging things out. I always seem to miss one or two. Sorry!

Reply:
happens us all